metricbeat/helper/server/http/http.go

```diff
@@ -66,8 +72,9 @@ func NewHttpServer(mb mb.BaseMetricSet) (server.Server, error) {
 	}
 
 	httpServer := &http.Server{
-		Addr:    fmt.Sprintf("%s:%d", config.Host, config.Port),
-		Handler: http.HandlerFunc(h.handleFunc),
+		Addr:      fmt.Sprintf("%s:%d", config.Host, config.Port),
```

Comment:
Not a new issue, but I think this should be using `net.JoinHostPort()` to ensure that IPv6 addresses get enclosed in brackets since net.Dialer expects them.